src/coreclr/jit/codegencommon.cpp

```diff
@@ -2404,8 +2406,16 @@ void CodeGen::genReportEH()
     // Tell the VM how many EH clauses to expect.
     compiler->eeSetEHcount(EHCount);
 
-    XTnum = 0; // This is the index we pass to the VM
+    struct EHClauseInfo
+    {
+        CORINFO_EH_CLAUSE clause;
+        EHblkDsc*         HBtab;
+    };
+
+    EHClauseInfo* clauses = new EHClauseInfo[compiler->compHndBBtabCount];
```

Comment:
This is allocating memory on the global heap

```suggestion
    EHClauseInfo* clauses = new (compiler, CMK_Codegen) EHClauseInfo[compiler->compHndBBtabCount];
```